From b9e1662ece1baaf57d141ad045e316dffcbf53a4 Mon Sep 17 00:00:00 2001
From: Carl George <carl@george.computer>
Date: Thu, 20 Jul 2023 00:01:26 -0500
Subject: [PATCH] Rename executable to blackbox-terminal

This is to avoid file conflicts with the blackbox window manager, which
is already packaged in several Linux distros.
---
 com.raggesilver.BlackBox.json            | 2 +-
 data/com.raggesilver.BlackBox.desktop.in | 2 +-
 src/meson.build                          | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/com.raggesilver.BlackBox.json b/com.raggesilver.BlackBox.json
index 86e0a5d..030c22f 100644
--- a/com.raggesilver.BlackBox.json
+++ b/com.raggesilver.BlackBox.json
@@ -3,7 +3,7 @@
   "runtime": "org.gnome.Platform",
   "runtime-version": "45",
   "sdk": "org.gnome.Sdk",
-  "command": "blackbox",
+  "command": "blackbox-terminal",
   "finish-args": [
     "--filesystem=host",
     "--filesystem=xdg-run/gvfsd",
diff --git a/data/com.raggesilver.BlackBox.desktop.in b/data/com.raggesilver.BlackBox.desktop.in
index 28f3b55..1d45a17 100644
--- a/data/com.raggesilver.BlackBox.desktop.in
+++ b/data/com.raggesilver.BlackBox.desktop.in
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Black Box
-Exec=blackbox
+Exec=blackbox-terminal
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;System;TerminalEmulator;
diff --git a/src/meson.build b/src/meson.build
index 8891f58..47d03f1 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -75,7 +75,7 @@ if get_option('blackbox_is_flatpak')
   add_project_arguments('-D', 'BLACKBOX_IS_FLATPAK', language: 'vala')
 endif
 
-executable('blackbox', blackbox_sources, config_header,
+executable('blackbox-terminal', blackbox_sources, config_header,
   vala_args: '--target-glib=2.50',  dependencies: blackbox_deps,
   install: true,
 )
-- 
GitLab
